pkg/odo/genericclioptions/context.go

```diff
@@ -35,218 +34,140 @@ type Context struct {
 // internalCxt holds the actual context values and is not exported so that it cannot be instantiated outside of this package.
 // This ensures that Context objects are always created properly via NewContext factory functions.
 type internalCxt struct {
-	ComponentContext    string
+	// project used for the command, either passed with the `--project` flag, or the current one by default
+	project string
+	// application used for the command, either passed with the `--app` flag, or the current one by default
+	application string
+	// component used for the command, either passed with the `--component` flag, or the current one by default
+	component string
+	// componentContext is the value passed with the `--context` flag
+	componentContext string
+	// outputFlag is the value passed with the `--output` flag
+	outputFlag string
+	// The path of the detected devfile
+	devfilePath string
+	// Kclient can be used to access Kubernetes resources
+	KClient kclient.ClientInterface
+	// Client can be used to access OpenShift resources
 	Client              *occlient.Client
-	command             *cobra.Command
-	Project             string
-	Application         string
-	cmp                 string
-	OutputFlag          string
-	KClient             kclient.ClientInterface
 	EnvSpecificInfo     *envinfo.EnvSpecificInfo
 	LocalConfigProvider localConfigProvider.LocalConfigProvider
 }
 
 // CreateParameters defines the options which can be provided while creating the context
 type CreateParameters struct {
-	Cmd                    *cobra.Command
-	DevfilePath            string
-	ComponentContext       string
-	IsNow                  bool
-	CheckRouteAvailability bool
+	cmd               *cobra.Command
+	componentContext  string
+	routeAvailability bool
+	devfile           bool
+	offline           bool
+	appIfNeeded       bool
+}
+
+func NewCreateParameters(cmd *cobra.Command) CreateParameters {
+	return CreateParameters{cmd: cmd}
+}
+
+func (o CreateParameters) CheckRouteAvailability() CreateParameters {
```

Comment:
**Optional:** Can we rename this method to something like `RequireRouteAvailability`? I understand your reasoning behind naming it `CheckRouteAvailability`, but I still find it a little non-intuitive. Intuitively I think it would be a function that checks if the route is available to use.